src/components/TicketModal/TicketModal.vue

```diff
@@ -0,0 +1,272 @@
+<template>
+  <div class="ticket-modal">
+    <div
+      class="ticket-modal-header"
+      :style="{ 'background-color': computedStatusColor }"
+    >
+      <div class="logo">
+        <IconSluggo :height="50" :width="50" />
+      </div>
+      <div
+        class="ticket-name"
+        :style="{ 'background-color': computedStatusColor }"
+      >
+        <div class="editableText">
+          <div v-if="!isEditing">
+            <span class="text" @click="enableEditing">
+              {{ ticketTitle }}
+            </span>
+          </div>
+          <div v-if="isEditing" @focusout="disableEditing">
+            <input
+              v-model="tempText"
+              class="input text-box"
+              v-bind:style="{ 'background-color': computedStatusColor }"
+              v-focus
+            />
+          </div>
+        </div>
+      </div>
+      <div class="editable-icon" v-if="shouldShowPencil">
+        <i class="bx bx-pencil"></i>
+      </div>
+      <div class="dropdown ticket-status" :class="statusDropdownClass">
+        <div class="dropdown-trigger" @click="toggleStatusDropdown">
+          <button
+            class="button is-primary"
+            :style="{ 'background-color': computedStatusColor }"
+            aria-haspopup="true"
+            aria-controls="dropdown-menu"
+          >
+            <span>{{ ticketStatus }}</span>
+            <span class="icon is-small">
+              <i class="bx bx-chevron-down"></i>
+            </span>
+          </button>
+        </div>
+        <div class="dropdown-menu">
+          <div class="dropdown-content">
+            <div
+              class="dropdown-item"
+              v-if="ticketStatus != 'To Do'"
+              @click="setTicketStatus('To Do')"
+            >
+              To Do
+            </div>
+            <div
+              class="dropdown-item"
+              v-if="ticketStatus != 'In Progress'"
+              @click="setTicketStatus('In Progress')"
+            >
+              In Progress
+            </div>
+            <div
+              class="dropdown-item"
+              v-if="ticketStatus != 'Done'"
+              @click="setTicketStatus('Done')"
+            >
+              Done
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+    <div class="ticket-modal-first-row columns">
+      <div class="dropdown column" :class="userDropdownClass">
+        <label class="ticket-field-label">Assigned to</label>
+        <div class="dropdown-trigger" @click="toggleUserDropdown">
```

Comment:
Seems like you're duplicating the bulma dropdown quite a bit. Consider creating a dropdown component which extracts the state for showing the dropdown and an event for when an option is selected.

Reply:
I tried placing most of the things in components but it got kind of complex figuring out how to communicate the selections between the two components, because I'm assuming the parent component will need to know the state of all the children to send the finalized record to the API. I'll look into it some more, though.

Reply:
For communicating state across a hierarchy, children with `emit` events when something occurs (ie a choice is selected). The event parameters can contain the thing that changes (ie a choice). Here's the documentation on it: https://v3.vuejs.org/guide/migration/emits-option.html#overview

Admittedly, it's been a little while since I've used events, but this seems to be the recommended way of going about it.